tools/cldr-apps/src/main/java/org/unicode/cldr/web/WebContext.java

```diff
@@ -719,14 +720,39 @@ void print(Throwable t) {
      */
     void redirect(String where) {
         try {
-            response.sendRedirect(where);
+            String port = getRedirectPort();
+            if (port != null) {
+                String url =
+                        request.getScheme() + "://" + request.getServerName() + ":" + port + where;
```

Comment:
how do you know what the scheme is? 

couldn't this all just pull from `CLDR_SURVEY_URL` in the settings?

Reply:
maybe -- I didn't try that; I figured, like the original code, use the scheme and server name from the existing request, and just fix the port since that's what was broken when using 8888 for nginx